test/core_tests.js

```diff
@@ -0,0 +1,138 @@
+               "body.impress-disabled is removed." );
+    assert.ok( document.body.classList.contains("impress-enabled"),
+               "body.impress-enabled is added." );
+    
+    var canvas = document.querySelector( "div#impress > div" );
+    assert.ok( !canvas.classList.contains("step") && canvas.id === "",
+               "Additional 'canvas' div inserted between div#impress root and steps." );
+    assert.equal( canvas.style.transform,
+                  "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)",
+                  "canvas.style.transform initialized correctly" );
+    assert.equal( canvas.style.transformOrigin,
+                  "left top 0px",
+                  "canvas.style.transformOrigin initialized correctly" );
+    assert.equal( canvas.style.transformStyle,
+                  "preserve-3d",
+                  "canvas.style.transformStyle initialized correctly" );
+    assert.equal( canvas.style.transitionDelay,
+                  "0ms",
+                  "canvas.style.transitionDelay initialized correctly" );
+    // impress.js default values tries to set this to 1000ms, I'm completely confused about why that's not actually set in the browser?
+    assert.equal( canvas.style.transitionDuration,
+                  "0ms",
+                  "canvas.style.transitionDuration initialized correctly" );
+    assert.equal( canvas.style.transitionProperty,
+                  "all",
+                  "canvas.style.transitionProperty initialized correctly" );
+    assert.equal( canvas.style.transitionTimingFunction,
+                  "ease-in-out",
+                  "canvas.style.transitionTimingFunction initialized correctly" );
+                  
+    assert.equal( document.documentElement.style.height,
+                  "100%",
+                  "documentElement.style.height is 100%" );
+    
+    // Steps initialization
+    var step1 = document.querySelector( "div#step-1" );
+    assert.equal( step1.style.position,
+                  "absolute",
+                  "Step position is 'absolute'." );
+
+    assert.ok( step1.classList.contains("active"),
+               "Step 1 has active css class." );
+  }
+});
+
+// Note: Here we focus on testing the core functionality of moving between
+// steps, the css classes set and unset, and events triggered.
+// TODO: more complex animations and check position, transitions, delays, etc...
+// Those need to be separate html files, and there could be several of them.
+//
+// For now this just contains a single transition in order to get QUnit committed.
+// Complete test coverage needs to be added later.
+QUnit.test( "Impress Core API - Work in progress", function( assert ) {
+  // impress.js itself uses event listeners to manipulate most CSS classes. 
+  // Wait a short while before checking, to avoid race. 
+  // (See assertStepEnterWrapper and assertStepLeaveWrapper.)
+  var wait = 5; // milliseconds
```

Comment:
I managed to remove the need for this timeout with the following code:

```javascript
var root = document.querySelector( "div#impress" );
var step2 = document.querySelector( "div#step-2" );
var done = assert.async();
root.addEventListener( "impress:stepenter",  function( event ) {
  assert.strictEqual( event.target, step2, "Triggered event for the second step" );
  done();
});
impress().next();
```

In this case it triggers the test correctly without having to use `setTimeout, is there anything I am missing here?